calendar.php

```diff
@@ -242,316 +242,229 @@
+        $select['repeats_5_month'][$mybb->get_input('repeats_5_month', MyBB::INPUT_INT)] = "selected=\"selected\"";
+        $select['repeats_5_years'] = $mybb->get_input('repeats_5_years', MyBB::INPUT_INT);
+        $select['repeats_5_occurance'][$mybb->get_input('repeats_5_occurance')] = "selected=\"selected\"";
+        $select['repeats_5_weekday'][$mybb->get_input('repeats_5_weekday', MyBB::INPUT_INT)] = "selected=\"selected\"";
+        $select['repeats_5_month2'][$mybb->get_input('repeats_5_month2', MyBB::INPUT_INT)] = "selected=\"selected\"";
+        $select['repeats_5_years2'] = $mybb->get_input('repeats_5_years2', MyBB::INPUT_INT);
+
+        $timezone = $mybb->get_input('timezone', MyBB::INPUT_INT);
+    } else {
+        if (!empty($mybb->input['day'])) {
+            $day = $mybb->get_input('day', MyBB::INPUT_INT);
+        } else {
+            $day = my_date("j");
+        }
+        $select['single_day'][$day] = $select['start_day'][$day] = $select['end_day'][$day] = "selected=\"selected\"";
+
+        if (!empty($mybb->input['month'])) {
+            $month = $mybb->get_input('month', MyBB::INPUT_INT);
+        } else {
+            $month = my_date("n");
+        }
+        $select['single_month'][$month] = $select['start_month'][$month] = $select['end_month'][$month] = "selected=\"selected\"";
+
+        if (!empty($mybb->input['year'])) {
+            $single_year = $start_year = $end_year = $mybb->get_input('year', MyBB::INPUT_INT);
+        } else {
+            $single_year = $start_year = $end_year = my_date("Y");
+        }
+
+        $select['start_time'] = $select['end_time'] = '';
+        $select['type_single'] = "checked=\"checked\"";
+        $select['type_ranged'] = '';
+        $select['type'] = "single";
+        $select['repeats_1_days'] = 1;
+        $select['repeats_3_weeks'] = 1;
+        $select['repeats_4_type'][1] = "checked=\"checked\"";
+        $select['repeats_4_day'] = 1;
+        $select['repeats_4_months'] = 1;
+        $select['repeats_4_occurance'][1] = "selected=\"selected\"";
+        $select['repeats_4_weekday'][0] = "selected=\"selected\"";
+        $select['repeats_4_months2'] = 1;
+        $select['repeats_5_type'][1] = "checked=\"checked\"";
+        $select['repeats_5_day'] = 1;
+        $select['repeats_5_month'][1] = "selected=\"selected\"";
+        $select['repeats_5_years'] = 1;
+        $select['repeats_5_occurance'][1] = "selected=\"selected\"";
+        $select['repeats_5_weekday'][0] = "selected=\"selected\"";
+        $select['repeats_5_month2'][1] = "selected=\"selected\"";
+        $select['repeats_5_years2'] = 1;
+        $timezone = $mybb->user['timezone'];
+    }
+
+    $years = [];
+
+    // Construct option list for years
+    for ($year = my_date('Y'); $year < (my_date('Y') + 5); ++$year) {
+        if ($year == $single_year) {
+            $select['single_year'] = "selected=\"selected\"";
```

Comment:
Would be nice to do this logic inside the view rather than building HTML in the backend code. Same goes with the other instances of this type of code.

Reply:
I agree, but I'm not sure how to do that.

Reply:
Two options:

- Do the conditional check in Twig
- Set a variable such as `$select['single_year_selected'] = true` and in twig do something like:
```twig
<option{%if select.single_year_selected %} selected="selected"{%endif}/>
```

Option two is the easiest by far.